src/logic/wallets/ethTransactions.ts

```diff
@@ -20,6 +26,44 @@ const fetchGasPrice = async (gasPriceOracle: GasPriceOracle): Promise<string> =>
   return gasPrice.toString()
 }
 
+export const setMaxPrioFeePerGas = (maxPriorityFeePerGas: number, maxFeePerGas: number): number => {
+  return maxPriorityFeePerGas > maxFeePerGas ? maxFeePerGas : maxPriorityFeePerGas
+}
+
+export const getFeesPerGas = async (): Promise<{
+  maxFeePerGas: number
+  maxPriorityFeePerGas: number
+}> => {
+  let blocks: FeeHistoryResult | undefined
+  let maxPriorityFeePerGas: number | undefined
+  let baseFeePerGas: number | undefined
+
+  const web3 = getWeb3ReadOnly()
+
+  try {
+    // Lastest block, 50th reward percentile
+    blocks = await web3.eth.getFeeHistory(1, 'latest', [50])
+
+    // hexToNumber can throw if not parsing a valid hex string
```

Comment:
I agree that it would clean things up but considering we started the week with the `NaN` oracle issue, I'd say it's better to keep the extra check to be on the safe side.